fedimint-atomic-broadcast/src/db.rs

```diff
@@ -0,0 +1,133 @@
+use std::io::Cursor;
+
+use fedimint_core::db::Database;
+use fedimint_core::encoding::{Decodable, Encodable};
+use fedimint_core::impl_db_record;
+
+use crate::SignedBlock;
+
+#[derive(Debug, Encodable, Decodable)]
+struct SignedBlockKey(u64);
+
+impl_db_record!(
+    key = SignedBlockKey,
+    value = SignedBlock,
+    db_prefix = 0x00,
+    notify_on_modify = false,
+);
+
+#[derive(Debug, Encodable, Decodable)]
+struct AlephBufferKey(u64, u64);
+
+impl_db_record!(
+    key = AlephBufferKey,
+    value = Vec<u8>,
+    db_prefix = 0x01,
+    notify_on_modify = false,
+);
+
+pub async fn load_block(db: &Database, index: u64) -> Option<SignedBlock> {
+    db.begin_transaction()
+        .await
+        .get_value(&SignedBlockKey(index))
+        .await
+}
+
+pub async fn open_session(db: Database, index: u64) -> (Cursor<Vec<u8>>, UnitSaver) {
+    let mut buffer = vec![];
+    let mut buffer_index = 0;
+    let mut transaction = db.begin_transaction().await;
```

Comment:
Style: we tend to name these `dbtx` to avoid confusion with a "consensus transaction" etc.

Reply:
Done